bot/exts/info/doc/_parsing.py

```diff
@@ -0,0 +1,328 @@
+            else:
+                element_markdown = markdown_converter.process_text(element)
+
+            element_markdown_length = len(element_markdown)
+            rendered_length += element_length
+            tag_end_index += element_markdown_length
+
+            if not element_markdown.isspace():
+                markdown_element_ends.append(tag_end_index)
+            result += element_markdown
+        else:
+            break
+
+    if not markdown_element_ends:
+        return ""
+
+    newline_truncate_index = find_nth_occurrence(result, "\n", max_lines)
+    if newline_truncate_index is not None and newline_truncate_index < _MAX_DESCRIPTION_LENGTH:
+        truncate_index = newline_truncate_index
+    else:
+        truncate_index = _MAX_DESCRIPTION_LENGTH
+
+    if truncate_index >= markdown_element_ends[-1]:
+        return result
+
+    markdown_truncate_index = max(cut for cut in markdown_element_ends if cut < truncate_index)
+    return result[:markdown_truncate_index].strip(_TRUNCATE_STRIP_CHARACTERS) + "..."
+
+
+def _create_markdown(signatures: Optional[List[str]], description: Iterable[Tag], url: str) -> str:
+    """
+    Create a markdown string with the signatures at the top, and the converted html description below them.
+
+    The signatures are wrapped in python codeblocks, separated from the description by a newline.
+    The result markdown string is max 750 rendered characters for the description with signatures at the start.
+    """
+    description = _get_truncated_description(
+        description,
+        markdown_converter=DocMarkdownConverter(bullets="•", page_url=url),
+        max_length=750,
+        max_lines=13
+    )
+    description = _WHITESPACE_AFTER_NEWLINES_RE.sub('', description)
+    if signatures is not None:
+        formatted_markdown = "".join(f"```py\n{signature}```" for signature in _truncate_signatures(signatures))
+    else:
+        formatted_markdown = ""
+    formatted_markdown += f"\n{description}"
+
+    return formatted_markdown
+
+
+def _class_filter_factory(class_names: Iterable[str]) -> Callable[[Tag], bool]:
+    """Create callable that returns True when the passed in tag's class is in `class_names` or when it's is a table."""
+    def match_tag(tag: Tag) -> bool:
+        for attr in class_names:
+            if attr in tag.get("class", ()):
+                return True
+        return tag.name == "table"
+
+    return match_tag
+
+
+def get_symbol_markdown(soup: BeautifulSoup, symbol_data: DocItem) -> Optional[str]:
+    """
+    Return parsed markdown of the passed symbol using the passed in soup, truncated to 1000 characters.
+
+    The method of parsing and what information gets included depends on the symbol's group.
+    """
+    symbol_heading = soup.find(id=symbol_data.symbol_id)
+    if symbol_heading is None:
+        log.warning("Symbol present in loaded inventories not found on site, consider refreshing inventories.")
```

Comment:
How much are things like warnings looked at and who has access to them? Wondering if this would be more suited as a message in some staff channel, it'd require some more logic around it as it'll most probably get logged multiple times when an updated page is fetched and there may be some false positives from weirdly structured docs

Reply:
Sentry would only create one issue for this and the Sentry webhook would only relay it once. It could be marked as resolved and later be automatically unresolved by Sentry as a regression. The raw logs are not looked at actively. They are looked at only rarely when something seems wrong and there's no other info. I'm not sure if this is adequate.